packages/clay-navigation-bar/src/index.tsx

```
@@ -0,0 +1,167 @@
+
+interface ItemProps extends React.HTMLAttributes<HTMLLIElement> {
+	/**
+	 * Determines the active state of an dropdown list item.
+	 */
+	active?: boolean;
+
+	/**
+	 * Children elements received from ClayNavigationBar.Item component.
+	 */
+	children: React.ReactElement;
+}
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Children elements received from ClayNavigationBar component.
+	 */
+	children: React.ReactElement<ItemProps>[];
+
+	/**
+	 * Determines the style of the Navigation Bar
+	 */
+	inverted?: boolean;
+
+	/**
+	 * Path to the location of the spritemap resource.
+	 */
+	spritemap: string;
+
+	/**
+	 * Set up dropdown's trigger label.
+	 */
+	triggerLabel: string;
+}
+
+type ItemType = React.FunctionComponent<ItemProps>;
+
+const Item: ItemType = ({active, children, className, ...otherProps}) => {
+	return (
+		<li
+			{...otherProps}
+			className={classNames('nav-item', {
+				active,
+				className,
+			})}
+		>
+			{React.Children.map(
+				children,
+				(child: React.ReactElement<ItemProps>, index) => {
+					return React.cloneElement(child, {
+						...child.props,
+						className: classNames(
+							'nav-link',
+							child.props.className,
+							{
+								active,
+							}
+						),
+						key: index,
+					});
+				}
+			)}
+		</li>
+	);
+};
+
+const ClayNavigationBar: React.FunctionComponent<Props> & {
+	Item: ItemType;
+} = ({
+	children,
+	className,
+	inverted = false,
+	spritemap,
+	triggerLabel,
+	...otherProps
+}) => {
+	const [visible, setVisible] = React.useState(false);
+	const contentRef = React.useRef<HTMLDivElement>(null);
+	const [
+		transitioning,
+		handleTransitionEnd,
+		handleClickToggler,
+	] = useTransition(visible, setVisible, contentRef);
+
+	const activeElementsCount = children.filter(child => child.props.active)
+		.length;
+
+	warning(
+		!(activeElementsCount > 1),
+		`You passed ${activeElementsCount} active props to ClayNavigationBar children.`
```

This warning would be more helpful if it indicated *what* the expected count is. eg. something like:

```javascript
warning(
  activeElementsCount <= 1,
  `ClayNavigationBar expects 0 or 1 active children, but received ${activeElementsCount}`
);
```
`<tangential-open-question>`I also wonder whether we want to fail a bit harder in cases like this, or in general (ie. by throwing an `Error` when APIs are used incorrectly). [They say](https://en.wikipedia.org/wiki/Robustness_principle) that you should be "liberal in what you accept, strict in what you provide", but they [also say](https://en.wikipedia.org/wiki/Fail-fast) "fail fast". In most situations, I find the latter approach to lead to higher quality.`</tangential-open-question>`

---

Makes sense! When creating this warning these principles wasn't in my mind. Thanks for guiding @wincent!